apm-agent-plugins/apm-servlet-plugin-it/simple-webapp-integration-test/src/test/java/co/elastic/apm/servlet/AbstractTomcatIntegrationTest.java

```diff
@@ -0,0 +1,153 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 the original author or authors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.servlet;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.ValidationMessage;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.logging.HttpLoggingInterceptor;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockserver.model.HttpRequest.request;
+
+/**
+ * When you want to execute the test in the IDE, execute {@code mvn clean package} before.
+ * This creates the {@code ROOT.war} file,
+ * which is bound into the docker container.
+ * <p>
+ * Whenever you make changes to the application,
+ * you have to rerun {@code mvn clean package}.
+ * </p>
+ * <p>
+ * To debug simple-webapp which is deployed to tomcat,
+ * add a break point in {@link #setUp()} and evaluate tomcatContainer.getMappedPort(8000).
+ * Then create a remote debug configuration in IntelliJ using this port and start debugging.
+ * TODO use {@link org.testcontainers.containers.SocatContainer} to always have the same debugging port
+ * </p>
+ */
+public abstract class AbstractTomcatIntegrationTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ServletIntegrationTest.class);
+
+    private static final String pathToWar = "../simple-webapp/target/ROOT.war";
```

Comment:
You need to give the full path to docker for mounting a directory. 

Example:
```
❯ docker run --rm -v ../shared:/app -w /app -ti alpine
docker: Error response from daemon: create ../shared: "../shared" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
```

Reply:
Assuming `withFileSystemBind` doesn't automatically expand the path for you.

Reply:
It does :)

Reply:
But let me check if we normalize it or not (we should)

Reply:
Just realised this works on travis ci, so obviously it must be somehow :)